Generate Client using OpenAPI spec tool,  Update Overview, Setup Guide and Quickstart (#2)
ballerina-platform/module-ballerinax-hubspot.marketing.subscriptions: ballerina/Package.md

## Overview

HubSpot is an AI-powered customer relationship management (CRM) platform. 

The `ballerinax/hubspot.marketing.subscriptions` offers APIs to connect and interact with the [HubSpot API for Marketing Subscriptions](https://developers.hubspot.com/docs/reference/api/marketing/subscriptions#get-%2Fcommunication-preferences%2Fv4%2Fstatuses%2F%7Bsubscriberidstring%7D) endpoints, specifically based on the [HubSpot Marketing Events API v3 OpenAPI spec](https://github.com/HubSpot/HubSpot-public-api-spec-collection/blob/13f888343a8ec443d95de2c12393015ba3391ac6/PublicApiSpecs/Communication%20Preferences/Subscriptions/Rollouts/176901/v4/subscriptions.json).

## Setup guide

> To use the HubSpot Marketing Subscriptions Connector, you must have access to the HubSpot API through a HubSpot developer account and a HubSpot App under it. Therefore you need to register for a developer account at HubSpot if you don't have one already.

Reviewer: ```suggestion
To use the HubSpot Marketing Subscriptions Connector, you must have access to the HubSpot API through a HubSpot developer account and a HubSpot App under it. Therefore, you need to register for a developer account at HubSpot if you don't have one already.
```

### Step 1: Create/Login to a HubSpot Developer Account

> If you have an account already, go to the [HubSpot developer portal](https://app.hubspot.com/)

Reviewer: ```suggestion
If you already have an account, go to the [HubSpot developer portal](https://app.hubspot.com/)
```

> If you don't have a HubSpot Developer Account you can sign up to a free account [here](https://developers.hubspot.com/get-started)

Reviewer: ```suggestion
If you don't have a HubSpot Developer Account, you can sign up to a free account [here](https://developers.hubspot.com/get-started)
```

### Step 2 (Optional): Create a [Developer Test Account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account

Within app developer accounts, you can create developer test accounts to test apps and integrations without affecting any real HubSpot data.

> **Note:** These accounts are only for development and testing purposes. In production you should not use Developer Test Accounts.

Reviewer: ```suggestion
**Note:** These accounts are intended solely for development and testing purposes and should not be used in production environments.
```

   <img src=../docs/setup/resources/Test7.png alt="Hubspot app creation 1 testacc3" style="width: 70%;">

### Step 5: Get your Client ID and Client Secret

- Navigate to the Auth section of your app. Make sure to save the provided Client ID and Client Secret.

   <img src=../docs/setup/resources/Test8.png alt="Hubspot app creation 1 testacc3" style="width: 70%;">

### Step 6: Setup Authentication Flow

Before proceeding with the Quickstart, ensure you have obtained the Access Token using the following steps:

1. Create an authorization URL using the following format:

   ```read
   https://app.hubspot.com/oauth/authorize?client_id=<YOUR_CLIENT_ID>&scope=<YOUR_SCOPES>&redirect_uri=<YOUR_REDIRECT_URI>
   ```

   Replace the `<YOUR_CLIENT_ID>`, `<YOUR_REDIRECT_URI>` and `<YOUR_SCOPES>` with your specific value.

    **NOTE:** If you are using a `localhost` redirect url, make sure to have a listener running at the relevant port before executing the next step.

2. Paste it in the browser and select your developer test account to intall the app when prompted.

   <img src=../docs/setup/resources/Test9.png alt="Hubspot app creation 1 testacc3" style="width: 70%;">

> 3. A code will be displayed in the browser. Copy the code.

Reviewer: ```suggestion
3. An authorization code appears in the browser. Copy the code.
```

4. Run the following curl command. Replace the `<YOUR_CLIENT_ID>`, `<YOUR_REDIRECT_URI`> and `<YOUR_CLIENT_SECRET>` with your specific value. Use the code you received in the above step 3 as the `<CODE>`.

   - Linux/macOS

     ```bash
     curl --request POST \
     --url https://api.hubapi.com/oauth/v1/token \
     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

>    This command will return a JSON response containing access_token and refresh_token.

Reviewer: ```suggestion
   This command returns a JSON response containing access_token and refresh_token.
```

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

> To use the `HubSpot Marketing Subcriptions` connector in your Ballerina application, update the `.bal` file as follows:

Reviewer: ```suggestion
To use the `HubSpot Marketing Subscriptions` connector in your Ballerina application, update the `.bal` file as follows:
```

### Step 1: Import the module

Import the `hubspot.marketing.subscription` module and `oauth2` module.

```ballerina
import ballerinax/hubspot.marketing.subscriptions as hsmsubscriptions;
import ballerina/oauth2;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials in the above steps as follows:

   ```toml
    clientId = <Client Id>
    clientSecret = <Client Secret>
    refreshToken = <Refresh Token>
   ```

2. Instantiate a `hsmsubscriptions:ConnectionConfig` with the obtained credentials and initialize the connector with it.

    ```ballerina 
    configurable string clientId = ?;
    configurable string clientSecret = ?;
    configurable string refreshToken = ?;

    final hsmsubscriptions:ConnectionConfig hsmsubscriptionsConfig = {
        auth : {
            clientId,
            clientSecret,
            refreshToken,
            credentialBearer: oauth2:POST_BODY_BEARER
        }
    };

    final hsmsubscriptions:Client hsmsubscriptions = check new (hsmsubscriptionsConfig, "https://api.hubapi.com/communication-preferences/v4");
    ```

### Step 3: Invoke the connector operation

> Now, utilize the available connector operations. A sample usecase is shown below.

Reviewer: ```suggestion
Now, utilize the available connector operations. A sample use case is shown below.
```

> ## Overview

Reviewer: Revisit this file with above comments
